Update hexchat_ssl skip login when SASL required

STARRY-S opened the pull request:
## Updated hexchat_ssl to skip login when SASL required
Skip login and send message in hexchat when SASL required

- Related ticket: https://progress.opensuse.org/issues/97679
- Needles:
  there are two needles were updated in openQA:
  https://openqa.suse.de/tests/8361288#step/hexchat_ssl/14 
  https://openqa.suse.de/tests/8361288#step/hexchat_ssl/18
- Verification run: 
  https://openqa.suse.de/tests/8367735 (without `XCHAT_URL` param)
  https://openqa.suse.de/tests/8367727 (with `XCHAT_URL` param)

lilyeyes commented on tests/x11/hexchat_ssl.pm:
```diff
@@ -29,6 +39,7 @@ sub run {
 
     if (my $url = get_var("XCHAT_URL")) {
         x11_start_program("$name --url=$url", target_match => "$name-main-window");
+        irc_send_message($name);
```
Since you added this line could you please offer the verification run for "XCHAT_URL"?

STARRY-S replied:
@lilyeyes hold on please, I'll run it later.

STARRY-S replied:
> Since you added this line could you please offer the verification run for "XCHAT_URL"?

Test failed when I added `XCHAT_URL` param,
I'll try to fix it later.

rfan1 replied:
@STARRY-S the failed case is expected, correct?

STARRY-S replied:
@lilyeyes @rfan1 The test code has been updated and
test passed with or without the `XCHAT_URL` parameter.

STARRY-S replied:
and here is the [verification run with XCHAT_URL](https://openqa.suse.de/tests/8367420#).

jouyingbin commented:
As I know there is new needles in this fix after I talked to you offline.
You can update the description as 
**Needles**: Already merged, or paste the new needles link.

STARRY-S commented:
> As I know there is new needles in this fix after I talked to you offline. You can update the description as **Needles**: Already merged, or paste the new needles link.

Updated the description of this pull request.

STARRY-S commented:
No, please hold this PR, and I will fix this tomorrow.

On Mon, Mar 21, 2022 at 17:48 rfan1 ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In tests/x11/hexchat_ssl.pm
> <https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14529#discussion_r830915448>
> :
>
> > @@ -29,6 +39,7 @@ sub run {
>
>      if (my $url = get_var("XCHAT_URL")) {
>          x11_start_program("$name --url=$url", target_match => "$name-main-window");
> +        irc_send_message($name);
>
> @STARRY-S <https://github.com/STARRY-S> the failed case is expected,
> correct?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14529#discussion_r830915448>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJ4GPW6CF3HP6MKFMZD3QRDVBBAWLANCNFSM5RG6HG6Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> <os-autoinst/os-autoinst-distri-opensuse/pull/14529/review/915472014@
> github.com>
>

lilyeyes reviewed (approved):
LGTM.

STARRY-S commented:
Found a new problem:
https://openqa.suse.de/tests/8367467#step/hexchat_ssl/14

hexchat exit before connect to the server, need fix.

jouyingbin commented:
@STARRY-S you forgot to remove WIP in the subject, it will show the CI check running.

STARRY-S commented:
> @STARRY-S you forgot to remove WIP in the subject, it will show the CI check running.

I found a new problem about this test case, and I'm working on to fix it, so I added the WIP again.

STARRY-S commented:
> > @STARRY-S you forgot to remove WIP in the subject, it will show the CI check running.
> 
> I found a new problem about this test case, and I'm working on to fix it, so I added the WIP again.

Fixed.

jouyingbin reviewed (approved):
LGTM, thanks for the patch.:)
